lib/index.js

```diff
@@ -8,6 +8,7 @@ var Spawn = require("child_process").spawn
   , Ul = require("ul")
   , OneByOne = require("one-by-one")
   , Sliced = require("sliced")
+  , Q = require('q')
```

Comment:
> In my opinion overhead of including and creating promise is really extremely small and it is better to keep API cleaner and simpler. 

Hmm, it's a [~2k lines file](https://github.com/kriskowal/q/blob/75058c0d70d36dd7e814c13493866043df39c858/q.js). Maybe is it possible to use a minimal implementation for promises? For example @RubenVerborgh's [`promiscuous`](https://github.com/RubenVerborgh/promiscuous). :sparkles:

Probably it has the same API, but it's few times smaller.